Source/Core/VideoCommon/VertexLoader.cpp

```diff
@@ -318,17 +318,17 @@ void VertexLoader::CompileVertexTranslator()
     m_native_vtx_decl.texcoords[i].type = VAR_FLOAT;
     m_native_vtx_decl.texcoords[i].integer = false;
 
-    const int format = m_VtxAttr.texCoord[i].Format;
-    const int elements = m_VtxAttr.texCoord[i].Elements;
+    const auto format = m_VtxAttr.texCoord[i].Format;
+    const auto elements = m_VtxAttr.texCoord[i].Elements;
 
     if (tc[i] != NOT_PRESENT)
     {
       ASSERT_MSG(VIDEO, DIRECT <= tc[i] && tc[i] <= INDEX16,
                  "Invalid texture coordinates!\n(tc[i] = %d)", (u32)tc[i]);
-      ASSERT_MSG(VIDEO, FORMAT_UBYTE <= format && format <= FORMAT_FLOAT,
-                 "Invalid texture coordinates format!\n(format = %d)", format);
-      ASSERT_MSG(VIDEO, 0 <= elements && elements <= 1,
-                 "Invalid number of texture coordinates elements!\n(elements = %d)", elements);
+      ASSERT_MSG(VIDEO, ComponentFormat::UByte <= format && format <= ComponentFormat::Float,
+                 "Invalid texture coordinates format!\n(format = %d)", (u32)format);
+      ASSERT_MSG(VIDEO, elements == TexComponentCount::S || elements == TexComponentCount::ST,
+                 "Invalid number of texture coordinates elements!\n(elements = %d)", (u32)elements);
```

Comment:
Note the `(u32)format` here; ideally I'd just let the enum formatter handle this, but there doesn't seem to be an fmt equivalent of ASSERT_MSG.